pkg/controllers/secretstore/glob/glob.go

```diff
@@ -0,0 +1,42 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Inspired from https://github.com/argoproj/argo-cd/tree/master/util/glob
+*/
+
+package glob
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/gobwas/glob"
+)
+
+func Match(log logr.Logger, pattern, text string) bool {
```

Comment:
Are there any specific reasons to choose Glob over Regex? We already use Regex elsewhere and considering its flexibility, I believe we should use Regex in this case.

Reply:
I was using glob as it is "simpler" in regard of the behavior as it is usually for prefix and suffix.

```yaml
conditions:
  namespacesGlobs:
    - teamone-* # (prefixed by teamone)
    - *-spark # (suffixed by spark)
```

For this kind of behavior some of the CNCF projects has choosen glob.

Kyverno:

```yaml
spec:
  rules:
  - name: no-LoadBalancer
    match:
      any:
      - resources:
          names: 
          - "prod-*"
          - "staging"
          kinds:
          - Service
```

ArgoCD:

```yaml
spec:
  destinations:
  - namespace: teamone-*
    server: '*'
```

However if you prefer I can switch back to regex

Reply:
I agree that glob is much simpler than regex, but it does not cover as many use cases as regex. On the other hand, as far as I know, there is nothing glob can express that regex can't. Considering that, I would choose regex in this case.

Reply:
I'll do the change